src/OrchardCore.Modules/OrchardCore.HealthChecks/Startup.cs

```diff
@@ -17,17 +21,38 @@ public Startup(IShellConfiguration shellConfiguration)
             _shellConfiguration = shellConfiguration;
         }
 
-        public override void Configure(IApplicationBuilder app, IEndpointRouteBuilder routes, IServiceProvider serviceProvider)
+        public override void ConfigureServices(IServiceCollection services)
         {
-            var healthCheckOptions = serviceProvider.GetService<IOptions<HealthChecksOptions>>().Value;
+            services.AddScoped<IHealthChecksResponseWriter, DefaultHealthChecksResponseWriter>();
+            services.AddHealthChecks();
```

Comment:
For info `AddHealthChecks()` registers `HealthCheckPublisherHostedService`, a hosted service that will never get called as we are at the tenant level, not so important but maybe better to just register the `DefaultHealthCheckService` singleton.

Reply:
> For info AddHealthChecks() registers HealthCheckPublisherHostedService

Right, but it doesn't work properly AFAIK, I tried many times, at the end I invoked `IHealthCheckService.Check.CheckHealthAsync()` to enforce register HC services to be triggered

> not so important but maybe better to just register the DefaultHealthCheckService singleton.

I might add it as an extension method on `OC.HealthAbstractions` while hosted service will no longer used

Reply:
> Right, but it doesn't work properly AFAIK

This what I said, a hosted service registered at the tenant level will never get called.

Reply:
Unfortunately `DefaultHealthCheckService` is an internal type :(, so I think it's fine to keep it as its

Reply:
Okay then